heritability/rhemc_pipeline.py

```diff
@@ -0,0 +1,1334 @@
+    return final_sink
+
+
+def get_num_bins(args):
+    """ Gets the number of bins that will be used in this analysis.
+    """
+    n_maf = 2 if args.maf_bins_2 else 5
+    n_ld = args.num_ld_bins
+    return n_maf * n_ld
+
+
+def get_num_iter(args):
+    """ Determines if iterations per phenotype must be performed (only enabled for testing).
+    If so, formats iterations properly for use.
+    """
+    if args.n_iter is None:
+        return [None]
+    else: 
+        return range(0, args.n_iter)
+
+
+def _get_pheno_manifest_path_internal():
+    return f'gs://ukb-diverse-pops/{loc}/phenotype_manifest.tsv.bgz'
+
+
+def _import_manifest():
+    manifest = hl.import_table(_get_pheno_manifest_path_internal())
+    manifest = manifest.annotate(pop_split = manifest.pops.split(','))
+    if 'phenotype_id' not in list(manifest.row):
+        manifest = manifest.annotate(phenotype_id = construct_phenotype_id(manifest))
+    manifest = manifest.annotate(pheno_file = get_pheno_filename(manifest.phenotype_id),
+                                 saige_file = convert_pheno_id_to_potential_saige(manifest.phenotype_id))
+    return manifest
+
+
+def _make_phenotype_dict(manifest, ancestries, n_include=None, random_phenos=None, suffix='', specific_pheno=None):
+    """ Internal function to construct a phenotype dictionary. Used to iterate
+    through phenotypes during pipeline construction.
+
+    Parameters
+    ----------
+    manifest: `Hail Table`
+    Imported manifest table. This function relies on modifications made to this table
+    in `_import_manifest`.
+
+    ancestries: `list`
+
+    n_include: `int` or None
+    This is primarily used for testing. If a non-None value is provided,
+    this function will trim the dictionary to include n_include items from each
+    trait type.
+
+    random_phenos: `int` or None
+    If not none, then will queue up a bunch of random phenotype jobs. Each of these will
+    be run on all phenotypes.
+
+    suffix: `str`
+    Appended to the end of the results file to search for.
+
+    specific_pheno: `list` or None
+    A list of formatted phenotypes to subset the manifest to. If any are not found, an error will be thrown.
+    """
+    if random_phenos is not None:
+        dct_out = {}
+        for i in range(1, random_phenos+1):
+            pheno_id = RANDOMPREF(i)
+            pheno_file = get_pheno_filename(pheno_id)
+            saige_file = convert_pheno_id_to_potential_saige(pheno_id)
+            dct_out.update({pheno_id: (pheno_file, saige_file, ancestries)})
+        return dct_out
+    else:
+        phenotype_id = manifest.phenotype_id.collect()
+        pheno_file = manifest.pheno_file.collect()
+        saige_file = manifest.saige_file.collect()
+        anc_vec = [[anc for anc in anclist if anc in ancestries] 
+                for anclist in manifest.pop_split.collect()]
+
+        pheno_to_len = {phen: len(anclist) for phen, _, _, anclist in zip(phenotype_id, pheno_file, saige_file, anc_vec)}
```

Comment:
Why zip all 4 if you're only using 2?

Reply:
Tweaked so `zipped_vals = zip(phenotype_id, pheno_file, saige_file, anc_vec)` comes before `pheno_to_len` as all 4 elements in `zipped_vals` was used for a later line also.